### PR TITLE
Create App. - Feature Bundle - Dec 2020

### PR DESCRIPTION
**Pull request by AleziaKurdis**
Content of this feature bundle:

**New "Tools" menu to the Create Application**

image

The previous "**Selection**" and "**Actions**" menu has been renamed "**Select**" and "**Edit**".
We have now all the actions regrouped in the 3 menus: "**Select**", "**Edit**" and "**Tools**".

Some changes coming with this:
1- The menu "**Teleport To Selected Entities**" has been moved from the "Select" menu to the new "Tools" menu.

2- The "Hotkey" "G" is now used to toggle the grid (previously it was to align it to the selection). It was making sense to have G (for Grid) to manage the visibility of this one.

3- New "Hotkey" has been added to control the other actions related to the Grid:
 "**Activate/Deactivate Snap to Grid**" can be triggered by clicking "H"
"**Align Grid to Selected Entities**" can be triggered by clicking "J"
 "**Align Grid to Avatar**" can be triggered by clicking "K"

4- The actions: "**Align Grid to Selected Entities**" and "**Align Grid to Avatar**" will turn on the grid if it was not visible. (Doing this imply that you want to have the grid to be visible). But you will still need to do "**Toggle Grid**" (**G**) to turn the grid off.
Note that "**Align Grid to Selected Entities**" will happen only If there is at least one entity selected.

**Main "Edit" menu**

The 2 "**Import Entities**" actions have been brought back to the main "**Edit**" menu. 
(Since there was no unanimity about their removal)

A **separator** has been added to regroup the Create App's menuItems.

image

### CODE REVIEW

**Comment by AleziaKurdis** [comment]
This commit is fully functional, This means that it can be tested.
But "Do Not Merge" now. 
I might add additional features in the next week.
or do readjustments depending some feedbacks?

File: scripts/system/create/edit.js [hunk 1]
```diff
@@ -1465,7 +1482,7 @@ function setupModelMenus() {
         menuItemName: MENU_ENTITY_LIST_DEFAULT_RADIUS,
         afterItem: MENU_SHOW_ZONES_IN_EDIT_MODE
     });
-
+
```
Comment (ctrlaltdavid) [inline scripts/system/create/edit.js:1485]: Unwanted space char.

File: scripts/system/create/edit.js [hunk 2]
```diff
@@ -2371,7 +2404,7 @@ var PropertiesTool = function (opts) {
         }
         var i, properties, dY, diff, newPosition;
         if (data.type === "update") {
-
+
```
Comment (ctrlaltdavid) [inline scripts/system/create/edit.js:2407]: Unwanted spaces.

File: scripts/system/create/entityList/html/js/entityList.js [hunk 3]
```diff
@@ -318,8 +330,18 @@ function loaded() {
         elSelectTopParent = document.getElementById("selecttopparent");
         elAddChildrenToSelection = document.getElementById("addchildrentoselection");
         elSelectFamily = document.getElementById("selectfamily");
-        elSelectTopFamily = document.getElementById("selecttopfamily");
+        elSelectTopFamily = document.getElementById("selecttopfamily");
```
Comment (ctrlaltdavid) [inline scripts/system/create/entityList/html/js/entityList.js:333]: Unwanted space.

File: scripts/system/create/entityList/html/js/entityList.js [hunk 4]
```diff
@@ -498,6 +524,42 @@ function loaded() {
             EventBridge.emitWebEvent(JSON.stringify({ type: "teleportToEntity" }));
             closeAllEntityListMenu();
         };
+        elSetCameraFocusToSelection.onclick = function () {
+            EventBridge.emitWebEvent(JSON.stringify({ type: "setCameraFocusToSelection" }));
+            closeAllEntityListMenu();
+        };
+        elToggleLocalWorldMode.onclick = function () {
+            EventBridge.emitWebEvent(JSON.stringify({ type: 'toggleSpaceMode' }));
```
Comment (ctrlaltdavid) [inline scripts/system/create/entityList/html/js/entityList.js:532]: Inconsistent use of `'` vs. `"` in this area.

File: scripts/system/libraries/gridTool.js [hunk 5]
```diff
@@ -163,6 +163,14 @@ Grid = function() {
         newPosition = Vec3.subtract(newPosition, { x: 0, y: SelectionManager.worldDimensions.y * 0.5, z: 0 });
         that.setPosition(newPosition);
     };
+    
+    that.moveToAvatar = function() {
+        var position = MyAvatar.getJointPosition("LeftFoot");
+        if (position.x === 0 && position.y === 0 && position.z === 0) {
```
Comment (ctrlaltdavid) [inline scripts/system/libraries/gridTool.js:169]: Better to use `0.0`s, I think, to indicate that they're real numbers rather than integeters.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-31dd724_bbde93f5.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-31dd724_bbde93f5.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-31dd724.tar_bbde93f5.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-31dd724.tar_bbde93f5.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-31dd724_bbde93f5.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-31dd724_bbde93f5.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Quest-Alpha-PR894-31dd724_bbde93f5.apk

**Comment by AleziaKurdis** [comment]
**Grid upgrade...**
The Grid is now using a **"local" Entity** instead of a **deprecated 3d Overlay**.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-740f7dd_670d832c.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-740f7dd.tar_670d832c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-740f7dd.tar_670d832c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-740f7dd_670d832c.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-740f7dd_670d832c.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-740f7dd_670d832c.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Quest-Alpha-PR894-740f7dd_670d832c.apk

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by AleziaKurdis** [comment]
You can remove the "**Do Not Merge**" tag.
That will be all for this PR.
I expected to address the Grid issues with the Local mode, but the expected behavior is far to be clear.

**Comment by Aitolda** [comment]
Everything seems to be working as intended.

**Comment by SilverfishVR** [comment]
PR894-740f7dd

Looks good 👍 👍  only issue I found is that the default width of the create window (in desktop) is too narrow to fit the new buttons:
Screenshot 2020-12-04 212220

Actually it is also too narrow for the Properties page to show properly so maybe the best solution is to just make the default wider?

**Comment by AleziaKurdis** [comment]
Is this the tablet mode in Desktop?
( I would have expected this to do as in VR .. and be displayed in landscape)

**Comment by AleziaKurdis** [comment]
Maybe open a bug for this. Since it was not introduced with this PR but with the previous one.

**Comment by AleziaKurdis** [comment]
**Create Tools window width (in Desktop mode)...**
In desktop mode only, the default windows width is now 750px instead of 490px.
which is closer than what we have in HMD (which is landscape display)
the 2 Import buttons (only in HMD) are now one under the other, in full width.
No change in HMD.
---
Please test and let me know if this is fine.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-2020ccc_536bf4ef.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-2020ccc.tar_536bf4ef.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-2020ccc_536bf4ef.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-2020ccc.tar_536bf4ef.gz

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Quest-Alpha-PR894-2020ccc_536bf4ef.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-2020ccc_536bf4ef.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR894/Vircadia-Alpha-PR894-2020ccc_536bf4ef.exe

**Comment by SilverfishVR** [comment]
looks good to me, mind you, that's on a UWQHD monitor but as long as it fits in a full HD monitor (so you can resize it) I thinks it's good

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**